packages/thirdweb/src/wallets/smart/index.ts

```diff
@@ -122,13 +127,10 @@ export async function connectSmartAccount(
     };
   }
 
-  const factoryAddress =
-    options.factoryAddress ??
-    getDefaultAccountFactory(options.overrides?.entrypointAddress);
   const sponsorGas =
     "gasless" in options ? options.gasless : options.sponsorGas;
 
-  if (await isZkSyncChain(chain)) {
+  if (isZksyncChain) {
```

Comment:
might be cleaner to just move this block at the very top

Reply:
My thinking was the paymaster and entrypoint address configs were still valid for ZKsync chains, but if that's not the case then yes the conditions can be moved around

Reply:
no entrypoint is only for non-zk chains, so makes sense to check zk chain first

Reply:
Great! In that case, I've pushed an update that simplifies the ZKsync chain check